src/traces/histogram/calc.js

```diff
@@ -180,10 +157,181 @@ module.exports = function calc(gd, trace) {
+
+    // all but the first trace in this group has already been marked finished
+    // clear this flag, so next time we run calc we will run autobin again
+    if(trace._autoBinFinished) {
+        delete trace._autoBinFinished;
+
+        return trace[binAttr];
+    }
+
+    // must be the first trace in the group - do the autobinning on them all
+    var traceGroup = getConnectedHistograms(gd, trace);
+    var autoBinnedTraces = [];
+
+    var minSize = Infinity;
+    var minStart = Infinity;
+    var maxEnd = -Infinity;
+
+    var autoBinAttr = 'autobin' + maindata;
+    var i, tracei, calendar, firstManual;
+
+
+    for(i = 0; i < traceGroup.length; i++) {
+        tracei = traceGroup[i];
+
+        // stash pos0 on the trace so we don't need to duplicate this
+        // in the main body of calc
+        var pos0 = tracei._pos0 = pa.makeCalcdata(tracei, maindata);
+        var binspec = tracei[binAttr];
+
+        if((tracei[autoBinAttr]) || !binspec ||
+                binspec.start === null || binspec.end === null) {
+            calendar = tracei[maindata + 'calendar'];
+            var cumulativeSpec = tracei.cumulative;
+
+            binspec = Axes.autoBin(pos0, pa, tracei['nbins' + maindata], false, calendar);
+
+            // adjust for CDF edge cases
+            if(cumulativeSpec.enabled && (cumulativeSpec.currentbin !== 'include')) {
+                if(cumulativeSpec.direction === 'decreasing') {
+                    minStart = Math.min(minStart, pa.r2c(binspec.start, 0, calendar) - binspec.size);
+                }
+                else {
+                    maxEnd = Math.max(maxEnd, pa.r2c(binspec.end, 0, calendar) + binspec.size);
+                }
+            }
+
+            // note that it's possible to get here with an explicit autobin: false
+            // if the bins were not specified. mark this trace for followup
+            autoBinnedTraces.push(tracei);
+        }
+        else if(!firstManual) {
+            // Remember the first manually set binspec. We'll try to be extra
+            // accommodating of this one, so other bins line up with these
+            // if there's more than one manual bin set and they're mutually inconsistent,
+            // then there's not much we can do...
+            firstManual = {
+                size: binspec.size,
+                start: pa.r2c(binspec.start, 0, calendar),
+                end: pa.r2c(binspec.end, 0, calendar)
+            };
+        }
+
+        // Even non-autobinned traces get included here, so we get the greatest extent
+        // and minimum bin size of them all.
+        // But manually binned traces won't be adjusted, even if the auto values
+        // are inconsistent with the manual ones (or the manual ones are inconsistent
+        // with each other).
+        //
+        // TODO: there's probably a weird case here where a larger bin pushes the
+        // start/end out, then it gets shrunk and doesn't make sense with the smaller bin.
```

Comment:
I don't mind not fixing this either. It would be nice to add a test case to document this behavior though.